### PR TITLE
useBlockSync: just testing without isControlled effect

### PR DESCRIPTION
**Pull request by ellatrix**
## What?

Just checking what tests will fail, see https://github.com/WordPress/gutenberg/pull/60967#pullrequestreview-2021871170

Edit: ok, this passed 🙂

Edit: but it was first introduced in #37484 (fixing #37361), which didn't add any e2e tests. So we'll have to check that.

## Why?

Reduce complexity

### CODE REVIEW

File: packages/block-editor/src/components/provider/use-block-sync.js [hunk 1]
```diff
@@ -186,23 +177,6 @@ export default function useBlockSync( {
...
-			isMounted.current = true;
-			return;
-		}
-
-		// When the block becomes uncontrolled, it means its inner state has been reset
-		// we need to take the blocks again from the external value property.
-		if ( ! isControlled ) {
-			pendingChanges.current.outgoing = [];
-			setControlledBlocks();
-		}
```
Comment (youknowriad) [inline packages/block-editor/src/components/provider/use-block-sync.js:203]: So now, I have a vague memory of the Navigation block switching between "not controlled" into "controlled" to support some legacy mode of that block. @draganescu might know more. I guess the testing instructions is to try to load that old navigation block markup in the editor and ensure that everything works as expected.

**Comment by ellatrix** [comment]
So the PR that introduced this fixed #37361. I tried to reproduce the original issue, but maybe I'm doing it wrong. Edited the page content, then undo, but the header template contents are still there. Unfortunately there's no e2e test I think.

**Review by youknowriad — APPROVED** [review]
LGTM

**Review by jsnajdr — APPROVED** [review]
Yes, IMO the effect is guaranteed to never be called. The only place where `isControlled` can go from `true` to `false` is the cleanup effect function of `useBlockSync`, when the hook is unmounting. And at that time of unmounting and cleanup, no new effect will be scheduled.

File: packages/block-editor/src/components/provider/use-block-sync.js (hunk 1 above)
Comment (draganescu) [reply packages/block-editor/src/components/provider/use-block-sync.js:203]: The navigation block can start as uncontrolled, when there are navigation inner blocks specified in the markup. When this uncontrolled state is updated by say adding a new navigation link, the content is saved to a CPT and the block switches to controlled.
EDIT:
I tested this PR and the functionality described above seems unaffected.

**Comment by ellatrix** [comment]
Whoops, forgot to change the commit title 🤦‍♀️

File: packages/block-editor/src/components/provider/use-block-sync.js (hunk 1 above)
Comment (jsnajdr) [reply packages/block-editor/src/components/provider/use-block-sync.js:203]: But the potentialy buggy part is when you hit Undo after this switch from uncontrolled to controlled. Then probably the navigation block should switch back to uncontrolled? And revert to the uncontrolled inner blocks specified in the markup?

That's what the title of #37484 (the PR that claims to fix #37361) says: "avoid undo issues"

Comment (draganescu) [reply packages/block-editor/src/components/provider/use-block-sync.js:203]: >  the navigation block should switch back to uncontrolled? 

I can't tell if that was ever the case? I don't even know if we want that, the uncontrolled version is _just a way_ for themes to include default navigation content for style reasons.

**Pull request merged**